File: models/object_model.py
from typing import Optional
from uuid import UUID, uuid4
from datetime import datetime, timezone
from beanie import Document
from pydantic import Field


class Object(Document):

    id: UUID = Field(default_factory=uuid4)

    title: str
    description: Optional[str] = None
    image_url: Optional[str] = None
    image_public_id: Optional[str] = None

    created_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc)
    )

    class Settings:
        name = "Objects"

File: backend/models/object-model.ts
export interface ObjectItem {
  id: string;
  title: string;
  description: string;
  image_url?: string;
  image_public_id?: string;
  created_at: Date;
}

export type ObjectItemCreate = Pick<ObjectItem, "title" | "description"> & {
  image: File;
};

export type ObjectItemImageUpdate = {
  object_id: string;
  image: File;
};
